bundles/org.openhab.binding.magentatv/src/main/resources/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,238 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="magentatv"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="receiver">
+		<label>MagentaTV Media Receiver</label>
+		<description>Represents a Telekom Media Receiver for MagentaTV</description>
+
+		<channel-groups>
+			<channel-group id="control" typeId="control">
+				<label>Control</label>
+			</channel-group>
+			<channel-group id="program" typeId="program">
+				<label>Program Information</label>
+			</channel-group>
+			<channel-group id="status" typeId="status">
+				<label>Play Status</label>
+			</channel-group>
+		</channel-groups>
+
+
+		<config-description>
+			<parameter name="udn" type="text">
+				<label>Unique Device Name</label>
+				<description>The UDN identifies the Media Receiver</description>
+				<required>true</required>
+			</parameter>
+			<parameter name="modelId" type="text">
+				<label>Receiver Model</label>
+				<description>Model type (MR4xx/ MR2xx)</description>
+				<required>true</required>
+				<options>
+					<option value="Auto">AUTO</option>
+					<option value="DMS_TPB">MR400</option>
+					<option value="MR401B">MR401B</option>
+					<option value="MR201">MR201</option>
+				</options>
+			</parameter>
+			<parameter name="ipAddress" type="text">
+				<label>Media Receiver's IP Address</label>
+				<description>IP Address of the receiver</description>
+				<required>true</required>
+				<context>network-address</context>
+			</parameter>
+			<parameter name="port" type="text">
+				<label>Receiver Port</label>
+				<description>Port address for UPnP</description>
+				<required>true</required>
+			</parameter>
+			<parameter name="accountName" type="text">
+				<label>Account Name</label>
+				<description>Credentials: Login name (xxx@t-online.de, same as for the Telekom Kundencenter)</description>
+				<required>true</required>
```

Comment:
No, this is not required. Only in case it is there, it will be used to automatically retrieve the userID.

Reply:
Should be made an advanced parameter.

Reply:
Def. not advanced, because that's the standard mode. User enters credentials, binding obtains the userID and removes the credentials.

required=true is the trick, that the thing config pops up while you are adding the thing

Reply:
> required=true is the trick

Not a valid trick. It simply is not required and in 99% of the time the parameter should also not be set as it is only needed to get hold of a really required parameter (userId), which could also be obtained in other ways.

Reply:
refactored, see comment below